core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -74,19 +75,21 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     AdminUtils.createTopic(zkUtils, topic, 1, 1)
     AdminUtils.deleteTopic(zkUtils, topic)
     TestUtils.verifyTopicDeletion(zkUtils, topic, 1, servers)
-    assertFalse("Topic metrics exists after deleteTopic", checkTopicMetricsExists(topic))
+    assertEquals("Topic metrics exists after deleteTopic", Set.empty, topicMetricGroups(topic))
   }
 
   @Test
   def testBrokerTopicMetricsUnregisteredAfterDeletingTopic() {
     val topic = "test-broker-topic-metric"
     AdminUtils.createTopic(zkUtils, topic, 2, 1)
-    createAndShutdownStep(topic, "group0", "consumer0", "producer0")
-    assertTrue("Topic metrics don't exist", checkTopicMetricsExists(topic))
+    // Produce a few messages to create the metrics
+    // Don't consume messages as it may cause metrics to be re-created causing the test to fail, see KAFKA-5238
```

Comment:
After the topic is deleted, I imagine the value being added to the metrics is zero in the consume case. So it may be simple to check for zero before updating metrics. But that can be done in KAFKA-5238.